docs: getting started (#514)
libp2p/js-libp2p: doc/API.md

console.log(peerStats.toJSON())
```

## Events

Once you have a libp2p instance, you are able to listen to several events it emits, so that you can be noticed of relevant network events.

#### An error has occurred

`libp2p.on('error', (err) => {})`

- `err`: instance of `Error`

#### A peer has been discovered

`libp2p.on('peer:discovery', (peer) => {})`

If `autoDial` option is `true`, applications should **not** attempt to connect to the peer
unless they are performing a specific action. See [peer discovery and auto dial](./PEER_DISCOVERY.md) for more information.

- `peer`: instance of [PeerInfo][https://github.com/libp2p/js-peer-info]

> #### We have a new connection to a peer

Reviewer: ```suggestion
#### An incoming connection from a peer has been established
```

Is it only incoming or outgoing also?

Reviewer 2: It is both.

Reviewer: Gotcha, might be an idea just to be explicit that it is!

`libp2p.on('peer:connect', (peer) => {})`

- `peer`: instance of [PeerInfo][https://github.com/libp2p/js-peer-info]

> #### We have closed a connection to a peer

Reviewer: ```suggestion
#### An existing connection to a peer has been closed
```

Can we explain why this might happen? Is it only if the remote disconnects or do we also get this event when we disconnect?

Reviewer 2: It occurs anytime we are disconnected from a peer, regardless of what caused it.
